app/controllers/api/subcollections/security_groups.rb

```diff
@@ -4,6 +4,30 @@ module SecurityGroups
       def security_groups_query_resource(object)
         object.respond_to?(:security_groups) ? object.security_groups : []
       end
+
+      def security_groups_add_resource(parent, _type, _id, data)
+        security_group = data["security_group"]
+
+        begin
+          message = "Adding security group #{security_group} to #{parent.name}"
+          task_id = queue_object_action(parent, message, :method_name => "add_security_group", :args => [security_group])
```

Comment:
I see the `query_resource` checks to see if it responds to `:security_groups` - I am assuming this means not all Instances have security groups, so should it be checking to see if the parent also responds to `add_security_group` before the action is queued? Same for the `remove_resource`